zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java

```diff
@@ -208,9 +208,6 @@ public Response putNotePermissions(@PathParam("noteId") String noteId, String re
     HashSet<String> writers = permMap.get("writers");
     // Set readers, if writers and owners is empty -> set to user requesting the change
     if (readers != null && !readers.isEmpty()) {
-      if (writers.isEmpty()) {
-        writers = Sets.newHashSet(SecurityUtils.getPrincipal());
```

Comment:
Thanks for good point!
This line `if` statement mean that when filling `readers` and not filling `writers`, it's filled with `writers` as current user. In this case, I think, it's enough for filling with `owners` as current user. 
And, the following screenshot and explain is in this PR description.
> Plus, if a user only set Readers role, all field(Writers and Owners) role is filled as the current user like below. It's not necessary to fill because the Owner role can read and write note.
https://cloud.githubusercontent.com/assets/8110458/24869746/2a77d2da-1e4f-11e7-86a5-794e1aac02d0.gif